pypeit/core/flux_calib.py

```diff
@@ -548,10 +549,81 @@ def sensfunc(wave, counts, counts_ivar, counts_mask, exptime, airmass, std_dict,
 
     return meta_table, out_table
 
+def get_sensfunc_factor(wave, wave_sens, sensfunc, exptime, telluric=None, extinct_correct=False,
+                         airmass=None, longitude=None, latitude=None, extrap_sens=False):
+    """
+    Get sensitivity function factor that will be multiplied into a spectrum. This code interpolates the sensitivity
+    function and can also multiply in extinction and telluric corrections.
+
+    FLAM, FLAM_SIG, and FLAM_IVAR are generated
+
+    Args:
+        wave (float ndarray): shape = (nspec,)
+           Senstivity
+        wave_sens (float ndarray):
+           Sensfunc wavelength vector shape = (nsens,)
+        sensfunc (float ndarray): shape = (nsens,)
+           Sensitivity function
+        exptime (float):
+        telluric_correct:
+        extinct_correct:
+        airmass (float, optional):
+        longitude (float, optional):
+            longitude in degree for observatory
+        latitude:
+            latitude in degree for observatory
+            Used for extinction correction
+        extrap_sens (bool, optional):
+            Extrapolate the sensitivity function (instead of crashing out)
+
+    Returns:
+        sensfunc_interp/exptime/delta_wave: shape = (nspec,)
+
+    """
+
+    # TODO Telluric corrections via this method are deprecated
+    # Did the user request a telluric correction?
+    if telluric is not None:
+        # This assumes there is a separate telluric key in this dict.
+        msgs.info('Applying telluric correction')
+        sensfunc = sensfunc * (telluric > 1e-10) / (telluric + (telluric < 1e-10))
+
+    sensfunc_obs = np.zeros_like(wave)
+    wave_mask = wave > 1.0  # filter out masked regions or bad wavelengths
+    delta_wave = wvutils.get_delta_wave(wave, wave_mask)
+    try:
+        sensfunc_obs[wave_mask] = interpolate.interp1d(wave_sens, sensfunc, bounds_error=True)(wave[wave_mask])
+    except ValueError:
+        if extrap_sens:
+            sensfunc_obs[wave_mask] = interpolate.interp1d(wave_sens, sensfunc, bounds_error=False)(wave[wave_mask])
+            msgs.warn(
+                "Your data extends beyond the bounds of your sensfunc. You should be adjusting "
+                "the par['sensfunc']['extrap_blu'] and/or par['sensfunc']['extrap_red'] to extrapolate further "
+                "and recreate your sensfunc. But we are extrapolating per your direction. Good luck!")
+        else:
+            msgs.error("Your data extends beyond the bounds of your sensfunc. " + msgs.newline() +
+                       "Adjust the par['sensfunc']['extrap_blu'] and/or par['sensfunc']['extrap_red'] to extrapolate "
+                       "further and recreate your sensfunc.")
+
+    if extinct_correct:
+        if longitude is None or latitude is None:
```

Comment:
I don't see how this can ever trip.  We must have 
known lon and lat for every spectrograph.

Reply:
Additionally, make them required arguments of this method.

Reply:
They should not be required, because extinct_correct is optional.

Reply:
It will trip if whomever calls the function does not pass in the longitude and latitude. This is a core function and so spectrograph is not a required argument.

Reply:
It trips if the user does not pass in longitude and latitude.